### PR TITLE
Use new libE save history options

### PR DESCRIPTION
**Pull request by AngelFP**
Use the new `libE_specs` options `"save_H_on_completion"`, `"save_H_with_date"` and `"H_file_prefix"` to customize the name of the history file and make sure that there is only one file per exploration.

### CODE REVIEW

**Review by jlnav — APPROVED** [review]
Looks good to me! Note that we plan on releasing libE v1.1 today if possible

**Comment by AngelFP** [comment]
> Looks good to me! Note that we plan on releasing libE v1.1 today if possible

Thanks, that's great news! I'll wait for the release before merging then so that we don't need to switch to `libensemble/develop`.

**Pull request merged**
